cortex/content.py

```diff
@@ -30,165 +34,221 @@
 
 class ManagedContentClient(_Client):
     """
-    A client used to access the Cortex managed content service (blob store).
-    """
-    URIs = {'content': 'projects/{projectId}/content'}
+    A client used to access the `Cortex managed content service (blob store) <https://cognitivescale.github.io/cortex-fabric/docs/manage-data/managed-content>`_. You can find a pre-created instance of this class on every :py:class:`cortex.client.Client` instance via the :py:attr:`Client.content` attribute.
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    >>> from cortex import Cortex; client = Cortex.client();
+    >>> client.content.list() # list content from the default project configured for the user
 
-    def upload(self, key: str, stream_name: str, stream: object, content_type: str, retries: int = 1):
-        """Store `stream` file in S3.
+    """  # pylint: disable=line-too-long
 
-        :param key: The path where the file will be stored.
-        :param stream_name: The name under which to save the `stream`.
+    URIs = {"content": "projects/{projectId}/content"}
+
+    def upload(
+        self,
+        key: str,
+        stream_name: str,
+        stream: Union[BytesIO, StringIO, FileIO],
+        content_type: str,
+        retries: int = 1,
+    ) -> dict:
+        """Store `stream` file in Managed Content (S3).
+
+        :param key: The path where the file will be stored in managed content
+        :type key: str
+        :param stream_name:  The name under which to save the `stream`.
+        :type stream_name: str
         :param stream: The file object.
+        :type stream: Union[BytesIO, StringIO, FileIO]
         :param content_type: the type of the file to store (e.g., `text/csv`).
-        :param retries: Number of times to retry a failed request from a retryable response.
+        :type content_type: str
+        :param retries: Number of times to retry a failed request from a retryable response, defaults to 1
+        :type retries: int, optional
         :return: a dict with the response to request upload.
+        :rtype: dict
 
         .. NOTE: This method uses a multi-part form request; to upload very large files, use `uploadStreaming` instead.
         .. seealso: uploadStreaming()
-        """
-        r = tenacity.Retrying(
+        """  # pylint: disable=line-too-long
+        res = tenacity.Retrying(
             stop=tenacity.stop_after_attempt(retries + 1),
```

Comment:
Kinda surprised we have user output here .. Retrying due to SSSSS; attempt N

Reply:
I'm going to add a generic retry state logger for all tenacity calls using the tenacity.Retrying(before=..) helper.